### PR TITLE
Add step to compare current page to a regex

### PR DESCRIPTION
**Pull request by Oursin**
Here is a proposal for a new builtin step that checks the page relative to a given regex.

The idea is that sometimes some parameters are located in the path of a page, such as an ID (`/user/123/profile`). In those cases, using a regex is much easier than trying to assert the exact value of those IDs, especially if the ID is random.

Adding a step `Verify page is like {{regex}}` will help in those cases.

**Changes**
Added `verifyAtPageRegex` function in `packages/js/browserinstance.js` that does the match against the given regex.
Added `Verify page is like {{regex}}` step in `packages/browser.smash`
Added tests in `tests/packages/browser.smash`

### CODE REVIEW

**Comment by faso** [comment]
Since the existing one is `Verify at page`, maybe this one should be `Very at page like {{regex}}`?

**Comment by Oursin** [comment]
It does seem better, i'll change that.
I think it would also be great to add regex alternatives to the `Wait until at page {{titleOrUrl}}` step, do you have any idea for a name ?

**Comment by faso** [comment]
That would be good for consistency I think. Let's see what @vptes1 thinks.

**Comment by vptes1** [comment]
This looks good! 

What do you think of merging this code with `Verify at page {{titleOrUrl}}`, such that if the title or url doesn't contain the given string, it tries parsing it as a regex? Just want to reduce the number of steps that are out there.

And I agree that we'd need corresponding `Wait until` and `Wait until...(up to {{n}} secs)` steps

**Comment by Oursin** [comment]
Thanks, I just merged the code, which now will support `Wait until` by default, since the same JS function is called.
I also updated the tests and added cases for the `Wait until` steps.

File: packages/js/browserinstance.js [hunk 1]
```diff
@@ -725,8 +725,12 @@ class BrowserInstance {
         try {
             await this.driver.wait(async () => {
                 obj = await this.executeScript(function(titleOrUrl) {
+                    let isMatched: document.title.toLowerCase().indexOf(titleOrUrl.toLowerCase()) != -1 || window.location.href.indexOf(titleOrUrl) != -1;
```
Comment (vptes1) [inline packages/js/browserinstance.js]: You mean `let isMatched =`?

Comment (Oursin) [reply packages/js/browserinstance.js]: It was a typo I made when copy-pasting my code, sorry. It should be fixed.

File: packages/js/browserinstance.js [hunk 2]
```diff
@@ -725,8 +725,12 @@ class BrowserInstance {
         try {
             await this.driver.wait(async () => {
                 obj = await this.executeScript(function(titleOrUrl) {
+                    let isMatched = document.title.toLowerCase().indexOf(titleOrUrl.toLowerCase()) != -1 || window.location.href.indexOf(titleOrUrl) != -1;
+                    let documentMatch = document.title.match(titleOrUrl);
+                    let locationMatch = window.location.href.match(titleOrUrl);
+                    isMatched = isMatched || documentMatch && documentMatch[0] === document.title || locationMatch && locationMatch[0] === window.location.href;
```
Comment (vptes1) [inline packages/js/browserinstance.js]: I'd put parenthesis around `documentMatch && documentMatch[0] === document.title` and `locationMatch && locationMatch[0] === window.location.href`. 

Also, maybe rename `documentMatch` to `titleRegexMatch` and `locationMatch` to `urlRegexMatch` or something like that.

Comment (Oursin) [reply packages/js/browserinstance.js]: Done !

File: tests/packages/browser.smash [hunk 3]
```diff
@@ -1203,19 +1218,36 @@ Open test page [
                 Wait until at page 'badstr'
                     - Verify error $s #manual
 
-        - Wait until at page (up to n secs)
```
[the remark this answers is not in the document]
Comment (Oursin) [reply tests/packages/browser.smash]: Woops, didn't see that. They are back.

**Pull request merged**